packages/react/src/components/DatePickerInput/next/DatePickerInput.js

```diff
@@ -0,0 +1,272 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import classNames from 'classnames';
+import { usePrefix } from '../../../internal/usePrefix';
+import {
+  Calendar,
+  WarningFilled,
+  WarningAltFilled,
+} from '@carbon/icons-react/next';
+
+const DatePickerInput = React.forwardRef(function DatePickerInput(
+  {
+    id,
+    labelText,
+    disabled = false,
+    invalid = false,
+    invalidText,
+    helperText,
+    hideLabel,
+    onChange = () => {},
+    onClick = () => {},
+    placeholder,
+    type = 'text',
+    datePickerType,
+    pattern = '\\d{1,2}\\/\\d{1,2}\\/\\d{4}',
+    iconDescription,
+    openCalendar,
+    size,
+    warn,
+    warnText,
+    ...other
+  },
+  ref
+) {
+  const prefix = usePrefix();
+
+  const datePickerInputProps = {
+    id,
+    onChange: (evt) => {
+      if (!disabled) {
+        onChange(evt);
+      }
+    },
+    onClick: (evt) => {
+      if (!disabled) {
+        onClick(evt);
+      }
+    },
+    placeholder,
+    type,
+    pattern,
+  };
+
+  const wrapperClasses = classNames(
+    `${prefix}--date-picker-input__wrapper NEXT-WRAPPER`,
```

Comment:
note for reviewers: the class `NEXT-WRAPPER` is just for testing purposes. To make sure it's using the next component. will remove before merging.